lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -101,91 +101,220 @@ export class BrowserStorage extends CacheManager {
+     * @param value
+     */
+    setRefreshTokenCredential(key: string, value: RefreshTokenEntity): void {
+        this.setItem(key, JSON.stringify(value));
+    }
+
+    /**
+     * fetch appMetadata entity from the platform cache
+     * @param key
+     */
+    getAppMetadata(key: string): AppMetadataEntity {
+        const value = this.getItem(key);
+        if (StringUtils.isEmpty(value)) {
+            return null;
         }
+        const appMetadata: AppMetadataEntity = CacheManager.toObject(new AppMetadataEntity(), JSON.parse(value));
+        if (AppMetadataEntity.isAppMetadataEntity(key, appMetadata)) {
+            return appMetadata;
+        }
+        return null;
+
+    }
+
+    /**
+     * set appMetadata entity to the platform cache
+     * @param key
+     * @param value
+     */
+    setAppMetadata(key: string, value: AppMetadataEntity): void {
+        this.setItem(key, JSON.stringify(value));
+    }
+
+    /**
+     * fetch server telemetry entity from the platform cache
+     * @param key
+     */
+    getServerTelemetry(key: string): ServerTelemetryEntity | null {
+        const value = this.getItem(key);
+        if (StringUtils.isEmpty(value)) {
+            return null;
+        }
+        const serverTelemetryEntity = CacheManager.toObject(new ServerTelemetryEntity(), JSON.parse(value));
+        if (ServerTelemetryEntity.isServerTelemetryEntity(key, serverTelemetryEntity)) {
+            return serverTelemetryEntity;
+        }
+        return null;
+    }
+
+    /**
+     * set server telemetry entity to the platform cache
+     * @param key
+     * @param value
+     */
+    setServerTelemetry(key: string, value: ServerTelemetryEntity): void {
+        this.setItem(key, JSON.stringify(value));
+    }
+
+    /**
+     * fetch throttling entity from the platform cache
+     * @param key
+     */
+    getThrottlingCache(key: string): ThrottlingEntity | null {
+        const value = this.getItem(key);
+        if (StringUtils.isEmpty(value)) {
+            return null;
+        }
+        const throttlingCache = CacheManager.toObject(new ThrottlingEntity(), JSON.parse(value));
+
+        if (ThrottlingEntity.isThrottlingEntity(key, throttlingCache)) {
+            return throttlingCache;
+        }
+        return null;
+    }
+
+    /**
+     * set throttling entity to the platform cache
+     * @param key
+     * @param value
+     */
+    setThrottlingCache(key: string, value: ThrottlingEntity): void {
+        this.setItem(key, JSON.stringify(value));
     }
 
     /**
      * Gets cache item with given key.
      * Will retrieve frm cookies if storeAuthStateInCookie is set to true.
      * @param key
      */
-    getItem(key: string, type: string): string | object {
-        const value = this.windowStorage.getItem(key);
+    getTemporaryCache(key: string): string {
+        if (this.cacheConfig.storeAuthStateInCookie) {
+            const itemCookie = this.getItemCookie(key);
+            return itemCookie;
```

Comment:
This should only return if `itemCookie` is truthy, otherwise, fall through to logic below.

Reply:
+1

Reply:
Done!